architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,259 @@
+
+If the TLS certificates and the private key used by the validating webhook are compromised, an attacker could impersonate the webhook server, intercepting the requests from the Kubernetes API Server or altering the responses. Another risk could be that the attacker could enable unauthorized clients to communicate with the webhook.
+
+**Impact**: The validating webhook functions return an approval or a validation error to the caller. The impersonator can return incorrect responses to the Kubernetes API Server which could affect the lifecycle of resources.
+
+Unauthorized clients wouldn't have any affect in the lifecycle of resources because, even if the webhook returns an approval or a rejection to the unauthorized client, the unauthorized client should also have access to the Kubernetes API Server to further trigger changes in the resource.
+
+**Mitigation**:
+
+1. **mTLS (Mutual TLS)**:
+
+   - **Description**: Ensure that mutual TLS is enabled for communication between the Validating Webhook and the Kubernetes API Server. This ensures that both parties authenticate each other, preventing unauthorized entities from impersonating the Controller.
+   - **Status**: Active
+
+2. **Strong Certificate Management**:
+
+   - **Description**: Implement strong certificate management practices, including regular rotation of certificates, using short-lived certificates, and monitoring for certificate anomalies. This reduces the risk of certificate compromise.
+   - **Status**:
+
+3. **Secure Storage**:
+
+   - **Description**: Store TLS certificates and private keys securely using secrets management solutions to prevent unauthorized access. Ensure that these secrets are encrypted and access is restricted to authorized components only.
+   - **Status**:
+
+4. **RBAC (Role-Based Access Control)**:
+
+   - **Description**: Implement strict RBAC policies to ensure that only authorized components and users have the necessary permissions to accessthe Validating Webhook (in this case; Kubernetes API Server). This minimizes the risk of unauthorized access.
+   - **Status**:
+
+5. **Audit Logs**:
+
+   - **Description**: Enable detailed audit logging for all interactions between the Validating Webhook and the Kubernetes API Server. Regularly review these logs to detect any unauthorized or suspicious activities.
+   - **Status**:
+
+6. **Network Policies**:
+
+   - **Description**: Implement necessary network policies to ensure that communication only from the Kubernetes API Server is accepted by the Validating Webhook.
+   - **Status**:
+
+#### Threat: Users with access to the webhook server modifying the behavior of the webhook server
+
+A user with access to the webhook server can modify its behavior to approve malicious requests, which can be reconciled in the next step of the data flow. Reconciliation loops can also trigger resource updates or even deletions by calling UCPD with the malicious requests.
```

Comment:
This is a great one for us to discuss in the context of assumptions.

Reply:
Assumptions: the user hasn't tampered with the Radius installation, Kubernetes cluster is not compromised.